Assets/Scripts/ViewManager.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using UnityEngine;
+using JueguitosPro.Views;
+
+namespace JueguitosPro
+{
+    public class ViewManager
+    {
+        private MainUI _mainUI;
+        
+        public ViewManager()
+        {
+            _mainUI = GameObject.FindObjectOfType<MainUI>();
+        }
+        
+        public T InstantiateView<T>(string prefabPath, MainUI.CanvasLayer canvasLayer, Action<T> callback = null) where T : ViewBase
+        {
+            RectTransform parent = _mainUI.GetCanvasLayer(canvasLayer);
+
+            var prefab = GetPrefabFromResources<T>(prefabPath);
+            
+            var view = GameObject.Instantiate(prefab);
```

Comment:
El var

Reply:
It is T, we said that for generic types is better to use var